federation/cmd/federation-controller-manager/app/controllermanager.go

```diff
@@ -92,17 +88,28 @@ func Run(s *options.CMServer) error {
 	} else {
 		glog.Errorf("unable to register configz: %s", err)
 	}
+
+	// If s.Kubeconfig flag is empty, try with the deprecated name in 1.4.
+	// TODO(madhusudancs): Remove this in 1.5.
+	var restClientCfg *restclient.Config
+	var err error
+	if len(s.Kubeconfig) <= 0 {
+		restClientCfg, err = restClientConfigFromSecret(s.Master)
+		if err != nil {
+			return err
+		}
+	}
+
 	// Create the config to talk to federation-apiserver.
-	kubeconfigGetter := util.KubeconfigGetterForSecret(KubeconfigSecretName)
-	restClientCfg, err := clientcmd.BuildConfigFromKubeconfigGetter(s.Master, kubeconfigGetter)
+	restClientCfg, err = clientcmd.BuildConfigFromFlags(s.Master, s.Kubeconfig)
```

Comment:
As discussed, this will overwrite the config.
Can remove what we are doing above.